lib/ansible/modules/cloud/vmware/vmware_host_kernel_manager.py

```diff
@@ -0,0 +1,221 @@
+    type: str
+  cluster_name:
+    description:
+    - Name of the VMware cluster to work on
+    - All ESXi hosts in this cluster will be configured
+    - This parameter is required if C(esxi_hostname) is not specified
+    type: str
+  kernel_module_name:
+    description:
+    - Name of the kernel module to be configured
+    required: true
+    type: str
+  kernel_module_option:
+    description:
+    - Specified configurations will be applied to the given module
+    - These values are specified in key=value pairs and separated by a space when there are multiple options
+    required: true
+    type: str
+extends_documentation_fragment: vmware.documentation
+'''
+
+EXAMPLES = '''
+- name: Configure IPv6 to be off via tcpip4 kernel module
+  vmware_host_kernel_manager:
+    hostname: '{{ vcenter_hostname }}'
+    username: '{{ vcenter_username }}'
+    password: '{{ vcenter_password }}'
+    esxi_hostname: '{{ esxi_hostname }}'
+    kernel_module_name: "tcpip4"
+    kernel_module_option: "ipv6=0"
+
+- name: Using cluster_name, configure vmw_psp_rr options
+  vmware_host_kernel_manager:
+    hostname: '{{ vcenter_hostname }}'
+    username: '{{ vcenter_username }}'
+    password: '{{ vcenter_password }}'
+    cluster_name: '{{ virtual_cluster_name }}'
+    kernel_module_name: "vmw_psp_rr"
+    kernel_module_option: "maxPathsPerDevice=2"
+'''
+
+RETURN = r'''
+results:
+    description:
+    - dict with information on what was changed, by ESXi host
+    returned: success
+    type: dict
+    sample: {
+    "results": {
+        "myhost01.example.com": {
+            "changed": true,
+            "configured_options": "ipv6=0",
+            "msg": "Options have been changed on the kernel module",
+            "original_options": "ipv6=1"
+        }
+    }
+}
+'''
+
+try:
+    from pyVmomi import vim, vmodl
+except ImportError:
+    pass
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.vmware import vmware_argument_spec, PyVmomi
+from ansible.module_utils._text import to_native
+
+
+class VmwareKernelManager(PyVmomi):
+    def __init__(self, module):
+        self.module = module
+        super(VmwareKernelManager, self).__init__(module)
+        cluster_name = self.params.get('cluster_name', None)
+        esxi_host_name = self.params.get('esxi_hostname', None)
+        self.hosts = self.get_all_host_objs(cluster_name=cluster_name, esxi_host_name=esxi_host_name)
+        self.kernel_module_name = self.params.get('kernel_module_name')
+        self.kernel_module_option = self.params.get('kernel_module_option')
+        self.results = {}
+
+        if not self.hosts:
+            self.module.fail_json(msg="Failed to find a host system that matches the specified criteria")
+
+    # find kernel module options for a given kmod_name. If the name is not right, this will throw an exception
+    def get_kernel_module_option(self, host, kmod_name):
+        host_kernel_manager = host.configManager.kernelModuleSystem
```

Comment:
Sometimes `ModuleSystem` is None so 
```python
if host_kernel_manager is None:
  self.module.fail_json(msg="Failed to find Kernel module system for %s" % host.name)
```

Reply:
Talked to akasurde on IRC, this is already checked in check_host_configuration_state which is called first before the get_kernel_module_option and apply_kernel_module_option functions.